R/Rules-class.R

```diff
@@ -173,108 +173,108 @@ NextBestThreePlusThree <- function() {
-##' \code{\linkS4class{DualEndpointEmax}})) However, is \code{scale} is set to
-##' "absolute" then the natural absolute biomarker scale can be used to set a target.
-##'
-##' @slot target the biomarker target range, that
-##' needs to be reached. For example, (0.8, 1.0) and \code{scale="relative"}
-##' means we target a dose
-##' with at least 80% of maximum biomarker level. As an other example,
-##' (0.5, 0.8) would mean that we target a dose between 50% and 80% of
-##' the maximum biomarker level.
-##' @slot scale either \code{relative} (default, then the \code{target} is interpreted
-##' relative to the maximum, so must be a probability range) or \code{absolute}
-##' (then the \code{target} is interpreted as absolute biomarker range)
-##' @slot overdose the overdose toxicity interval (lower limit excluded, upper
-##' limit included)
-##' @slot maxOverdoseProb maximum overdose probability that is allowed
-##' @slot targetThresh which target probability threshold needs to be fulfilled before the
-##' target probability will be used for deriving the next best dose (default: 0.01)
-##'
-##' @example examples/Rules-class-NextBestDualEndpoint.R
-##' @export
-##' @keywords classes
-.NextBestDualEndpoint <-
-    setClass(Class="NextBestDualEndpoint",
-             representation(target="numeric",
-                            scale="character",
-                            overdose="numeric",
-                            maxOverdoseProb="numeric",
-                            targetThresh="numeric"),
-             prototype(target=c(0.9,1),
-                       scale="relative",
-                       overdose=c(0.35, 1),
-                       maxOverdoseProb=0.25,
-                       targetThresh=0.01),
-             contains=list("NextBest"),
-             validity=
-                 function(object){
-                     o <- Validate()
+## class ----
 
-                     o$check(is.scalar(object@scale) && object@scale %in% c("relative", "absolute"),
-                             "scale must be either 'relative' or 'absolute'")
-                     if(object@scale == "relative")
-                     {
-                       o$check(is.probRange(object@target),
-                               "target has to be a probability range when scale='relative'")
-                     } else {
-                       o$check(is.range(object@target),
-                               "target must be a numeric range")
-                     }
-                     o$check(is.probRange(object@overdose),
-                             "overdose has to be a probability range")
-                     o$check(is.probability(object@maxOverdoseProb),
-                             "maxOverdoseProb has to be a probability")
-                     o$check(is.probability(object@targetThresh),
-                             "targetThresh has to be a probability")
+#' `NextBestDualEndpoint`
+#'
+#' @description `r lifecycle::badge("stable")`
```

Comment:
actually we should set all DualEndpoint classes as `experimental`, that is more honest...

Reply:
Sure, Thanks @danielinteractive!